### PR TITLE
[BACKLOG-38674] Integration Tests are outdated and some do not compile.

### PR DESCRIPTION
**Pull request by smmribeiro**
While, under BACKLOG-37696, the Rest step was moved to the Rest plugin, this code was "left behind".
I've moved it into the plugin tree.
Note that it is still needed to update the integration tests code to the new implementation!

@bcostahitachivantara @renato-s @andreramos89

### CODE REVIEW

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 50m 47s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl engine,plugins/rest/core
```
---
:ok_hand: All tests passed!
 
Tests run: **4101**, Failures: **0**, Skipped: **13** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9040/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by bcostahitachivantara — APPROVED** [review]

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 50m 38s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl \
engine,plugins/pur/core,plugins/rest/core
```
---
:ok_hand: All tests passed!
 
Tests run: **4101**, Failures: **0**, Skipped: **13** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9040/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: plugins/rest/core/smr.txt
Comment (bcostahitachivantara) [inline plugins/rest/core/smr.txt:1]: Why do not put this file in .gitignore?

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 50m 45s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl \
engine,plugins/pur/core,plugins/rest/core
```
---
:ok_hand: All tests passed!
 
Tests run: **4092**, Failures: **0**, Skipped: **13** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9040/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: plugins/rest/core/smr.txt
Comment (smmribeiro) [reply plugins/rest/core/smr.txt:1]: Good catch: it is not supposed to go in :-)

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 51m 32s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl \
engine,plugins/pur/core,plugins/rest/core
```
---
:ok_hand: All tests passed!
 
Tests run: **4092**, Failures: **0**, Skipped: **13** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9040/4/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Pull request merged**